packages/styled-components/src/utils/stringifyRules.js

```diff
@@ -0,0 +1,12 @@
+// @flow
+import isFunction from './isFunction';
+
+export default function stringifyRules(rules: Array<any>): string {
+  const newRules = rules.map(rule => {
+    if (isFunction(rule)) {
```

Comment:
This unfortunately is not SSR-safe. Code can differ between server and client depending on what babel flags are enabled etc.

Reply:
Do you mean `isFunction` or `rule.toString`?  It just use to generate id, or just try to add `displayName` for `GlobalStyleComponent` to generate different id ?

Reply:
> Do you mean isFunction or rule.toString?

`rule.toString()`, there can be subtle differences

> maybe we can try to add displayName for GlobalStyleComponent to generate different id like withConfig in StyledComponent?

That's possible but requires a bunch of architectural changes because cGS doesn't have the same construction pattern as `styled()`.

Reply:
Or，just try to generate different id for different function: 

```
const fnIDMap = new WeakMap<Function, number>();
let fnCount = 1;

function getId(fn: Function) {
  let fnId = fnIDMap.get(fn);
  if (!fnId) {
    fnId = fnCount ++;
    fnIDMap.set(fn, fnId);
  }

  return `fnId-${fnId}`;
}

export default function stringifyRules(rules: Array<any>): string {
  const newRules = rules.map(rule => {
    if (isFunction(rule)) {
      return getId(rule);
    }
    return rule;
  });
  return JSON.stringify(newRules);
}
```

Reply:
OK，I've got it. In our project, we try another way, just add id in comment:

```
const Style1 = createGlobalStyle`
// Id of this global style, don't remove
// Id `my global style name`
${props => css`
  .test {
    color: #999;
  }
`}
`;
```